cpu/sam0_common/periph/i2c.c

```diff
@@ -35,7 +37,7 @@
 #include "debug.h"
 
 /* guard file in case no I2C device is defined */
-#if I2C_NUMOF
+#ifdef I2C_NUMOF
```

Comment:
if it's defined as 0 line 62 will fail to compile, but it's still a valid value, meaning disabled.

Reply:
This issue can also be applied on sam0 UART. (maybe somewhere else too ?)
What should I do ?

Reply:
You are right, there's currently a mixture of `#if` and `#ifdef`s. It should be addressed by the build system in the future anway. So by now, whatever it's Ok

Reply:
Yes, I was thinking of opening an issue for this point to have community's opinion as  some driver use #if xxx_NUMOF and others #ifdef xxx_NUMOF. I'll do it later today.